### PR TITLE
javascript-javascript1-week1/NicoQ

### PR DESCRIPTION
**Pull request by lnknfan90**
Finished my HW

### CODE REVIEW

File: 3_javascript/javascript1/week1/Goodboy-Oldboy/script.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const dogYearFuture= 2060;
+
+//calculation
+const dogYear = dogYearFuture - dogYearOfBirth;
+
+// dog or not dog
+const shouldShowResultInDogYears = true ;
+
+// Answer
+if(shouldShowResultInDogYears){ console.log("Your dog will be " + (dogYear * 7) + " dog years old in " + dogYearFuture)}
```
Comment (elilovna) [inline 3_javascript/javascript1/week1/Goodboy-Oldboy/script.js:13]: try to use template literals for console.log and code alignments: 
if (shouldShowResultInDogYears) { 
    console.log("Your dog will be " + (dogYear * 7) + " dog years old in " + dogYearFuture)}
}

File: 3_javascript/javascript1/week1/Housey pricey/script.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  
+  if (peterHouse.cost < peterHousePrice) {
+    console.log("Peter is making a great deal!!");
+  } else if (peterHouse.cost > peterHousePrice) {
+    console.log("Peter is paying too much for the house.");
+  } else {
+    console.log("Peter is paying the expected price for the house.");
+  }
+  
+  console.log("");
```
Comment (elilovna) [inline 3_javascript/javascript1/week1/Housey pricey/script.js:43]: do you need this console?

File: 3_javascript/javascript1/week1/script.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
+
+let yearOfBirth = 1982;
```
Comment (elilovna) [inline 3_javascript/javascript1/week1/script.js:2]: it is better to use const if the value does not change.

could you please check all variables and change it to const if the value does not change

File: 3_javascript/javascript1/week1/script.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+//Dog year of birth
+let dogYearOfBirth = 2017;
+let dogYearFuture = 2027;
+let humanYears = dogYearFuture - dogYearOfBirth; 
+let dogYear = humanYears*7; 
+let shouldShowResultInDogYears = true;
+
+if (shouldShowResultInDogYears) {
+let dogAgeInDogYear = dogYear;
```
Comment (elilovna) [inline 3_javascript/javascript1/week1/script.js:15]: Could you please align the code here ?

File: 3_javascript/javascript1/week1/script.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+else {
+    console.log(`Your dog will be ${humanYears} human years old in ${dogYearFuture}`);
+}
+
+//House price generator for Your friend Peter is considering a house that is 8m wide, 
+//10m deep and 10m high. The garden size is 100m2. The house costs 2.500.000. 
+//Your friend Julia is considering a house that is 5m wide, 11m deep and 8m high. 
+//The garden size is 70m2. This house costs 1.000.000.
+//Formula is housePrice = volumeInMeters * 2.5 * 1000 + gardenSizeInM2 * 300;
+let houseDetails = [
```
Comment (elilovna) [inline 3_javascript/javascript1/week1/script.js:28]: cool 😎 that you used an array here; 

better to use const if you are not planing to change original array

File: 3_javascript/javascript1/week1/script.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+function calculateHousePrice(house) {
+    return house.width*house.depth*house.height*2.5*1000 + house.gardenSizeInM2*300;
+}
+function dealOrNoDeal(house) {
+    const housePrice = calculateHousePrice(house);
+    if (housePrice>house.houseCost) {
+        console.log(`${house.name} is being ripped off by the seller`);
+    }
+    else
+    {
```
Comment (elilovna) [inline 3_javascript/javascript1/week1/script.js:56]: code alignment:
} else {

File: 3_javascript/javascript1/week1/script.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    name:"Julia",
+    width:5,
+    depth:11,
+    height:8,
+    gardenSizeInM2:70,
+    houseCost:1000000,
+    }    
+];
+
+function calculateHousePrice(house) {
```
Comment (elilovna) [inline 3_javascript/javascript1/week1/script.js:47]: oh wow, cool that you tried to split the logic!

File: 3_javascript/javascript1/week1/script.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    const housePrice = calculateHousePrice(house);
+    if (housePrice>house.houseCost) {
+        console.log(`${house.name} is being ripped off by the seller`);
+    }
+    else
+    {
+        console.log(`${house.name} is getting a life changing deal`);
+    }
+}
+houseDetails.forEach(dealOrNoDeal);
```
Comment (elilovna) [inline 3_javascript/javascript1/week1/script.js:60]: Array methods 🥳 
supper cool

File: 3_javascript/javascript1/week1/script.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+//     const peterHousePrice =
```
Comment (elilovna) [inline 3_javascript/javascript1/week1/script.js:62]: if this code is not needed better to remove it

File: 3_javascript/javascript1/week1/script.js [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    "Widow",
+    "Tales",
+    "Rider",
+    "Dudes",
+    "Awakening",
+    "Laassi",
+    "Gringa",
+    "Dandy",
+];
+const startupName = firstWords[4] + " " + lastWords[0];
```
Comment (elilovna) [inline 3_javascript/javascript1/week1/script.js:129]: you could try to use Math.floor(Math.random() * 10 here :) to show different result

**Review by elilovna — CHANGES_REQUESTED** [review]
Outstanding job! 🌟

I can see that you've put in a lot of effort and made significant progress in creating variables and assigning values to them. 😎 I'm impressed by your ability to separate logic and use array methods, even though we haven't covered them yet. Bravo! 🥳

Here are a couple of friendly suggestions to make your code even better:

Whenever a variable's value won't change, make it a habit to use const.
Pay attention to code alignment for improved readability. Keep it up! 👍

File: 3_javascript/javascript1/week1/script.js (hunk 3 above)
Comment (lnknfan90) [reply 3_javascript/javascript1/week1/script.js:2]: Hello! I have noticed that I have made a mistake with the name of my branch, and I find two files that are not mine, one is this script.js and another is index.html in the week1 folder. I separated my files into individual folders per task. Should I create a new branch with my name so as not to overwrite anything? Nico Q

**Comment by shpomp** [comment]
image

**Pull request closed without merging**